blocks/rich-text/index.js

```diff
@@ -405,27 +408,13 @@ export default class RichText extends Component {
 	}
 
 	/**
-	 * Calculates the relative position where the link toolbar should be.
-	 *
-	 * Based on the selection of the text inside this element a position is
-	 * calculated where the toolbar should be. This can be used downstream to
-	 * absolutely position the toolbar. It does this by finding the closest
-	 * relative element.
+	 * Calculates relative position for a link modal in the formatting toolbar.
 	 *
-	 * @return {{top: number, left: number}} The desired position of the toolbar.
+	 * @return {{top: number, left: number}} Link modal position.
 	 */
 	getFocusPosition() {
 		const position = this.getEditorSelectionRect();
-
-		// Find the parent "relative" or "absolute" positioned container
-		const findRelativeParent = ( node ) => {
-			const style = window.getComputedStyle( node );
-			if ( style.position === 'relative' || style.position === 'absolute' ) {
-				return node;
-			}
-			return findRelativeParent( node.parentNode );
-		};
-		const container = findRelativeParent( this.editor.getBody() );
+		const container = this.editor.getBody().closest( '.edit-post-visual-editor' );
```

Comment:
The previous implementation was intentionally agnostic to its rendering context, since there's nothing that requires that `RichText` be rendered within the post editor (this is also why it was split out as a separate `edit-post` module).

Does this really depend on the post editor container, or is there another way we can express the intended logic which is less specific but equally effective?

In general, a component should have no awareness of where it is rendered. `Element#closest` in a component is definitely a code smell.

Reply:
> Does this really depend on the post editor container

Yes, it does depend on the visual editor now. That's a side-effect of the new [`VisualEditor.LastChild`](https://github.com/WordPress/gutenberg/pull/4503/files#diff-98010d4f3d9bc8b6dc2b7f26efbd75d8R42) slot, which is where the link modal lives in this PR.

I really didn't want to remove the `findRelativeParent()` function, but at the same time, it was not longer needed given the dependence on the visual editor container that we can target specifically.